app/domain/authentication/authn_oidc/authenticate_id_token/decode_and_verify_id_token.rb

```diff
@@ -41,12 +42,24 @@ def decoded_attributes
         def decoded_id_token
           @decoded_id_token ||= OpenIDConnect::ResponseObject::IdToken.decode(
             @id_token_jwt,
-            get_cert(@provider_uri)
+            get_cert
           )
+        rescue => e
+          raise IdTokenInvalidFormat, e.to_s
+        end
+
+        def discovered_provider
+          @discovered_provider ||= OpenIDConnect::Discovery::Provider::Config.discover!(@provider_uri)
```

Comment:
The `discover!` method being called here should not have a `!`. In Ruby, the `!` on a method name signals that it's a mutator method that modifies the object that calls it. I'm guessing that's not happening here, though correct me if I'm wrong!

Reply:
Also if you  make the change I suggested above to the `call` method, I don't think you need to memoize `@ discovered_provider` and can just return it from the method for use in `get_cert`.

Reply:
the `discover!` method is in the 3rd party

Reply:
Oh weird, okay!